tabpy-server/tabpy_server/handlers/base_handler.py

```diff
@@ -47,23 +53,230 @@ def _add_CORS_header(self):
         origin = self.tabpy_state.get_access_control_allow_origin()
         if len(origin) > 0:
             self.set_header("Access-Control-Allow-Origin", origin)
-            logger.debug("Access-Control-Allow-Origin:{}".format(origin))
+            logger.debug(self.append_request_context(
+                "Access-Control-Allow-Origin:{}".format(origin)))
 
         headers = self.tabpy_state.get_access_control_allow_headers()
         if len(headers) > 0:
             self.set_header("Access-Control-Allow-Headers", headers)
-            logger.debug("Access-Control-Allow-Headers:{}".format(headers))
+            logger.debug(self.append_request_context(
+                "Access-Control-Allow-Headers:{}".format(headers)))
 
         methods = self.tabpy_state.get_access_control_allow_methods()
         if len(methods) > 0:
             self.set_header("Access-Control-Allow-Methods", methods)
-            logger.debug("Access-Control-Allow-Methods:{}".format(methods))
+            logger.debug(self.append_request_context(
+                "Access-Control-Allow-Methods:{}".format(methods)))
 
     def _sanitize_request_data(self, data, keys=KEYS_TO_SANITIZE):
         """Remove keys so that we can log safely"""
         for key in keys:
             data.pop(key, None)
 
+    def _get_auth_method(self, api_version) -> (bool, str):
+        '''
+        Finds authentication method if provided.
+
+        Parameters
+        ----------
+        api_version : str
+            API version for authentication.
+
+        Returns
+        -------
+        bool
+            True if known authentication method is found.
+            False otherwise.
+
+        str
+            Name of authentication method used by client.
+            If empty no authentication required.
+
+        (True, '') as result of this function means authentication
+        is not needed.
+        '''
+        if api_version not in self.settings[SettingsParameters.ApiVersions]:
+            logger.critical(f'Unknown API version "{api_version}"')
+            return False, ''
+
+        version_settings =\
+            self.settings[SettingsParameters.ApiVersions][api_version]
+        if 'features' not in version_settings:
+            logger.info(f'No features configured for API "{api_version}"')
+            return True, ''
+
+        features = version_settings['features']
```

Comment:
nit: as we reference these hard coded strings in multiple places we should probably introduce an InfoParameters class like you did with SettingsParameters